griptape/drivers/video_generation/dream_machine_video_generation_driver.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import logging
+import time
+from typing import TYPE_CHECKING, Any
+
+import requests
+from attrs import Factory, define, field
+
+from griptape.artifacts import VideoArtifact
+from griptape.drivers import BaseVideoGenerationDriver
+from griptape.utils import import_optional_dependency
+
+if TYPE_CHECKING:
+    from lumaai import LumaAI
+
+logger = logging.getLogger(__name__)
+
+
+@define
+class DreamMachineVideoGenerationDriver(BaseVideoGenerationDriver):
+    api_key: str = field(kw_only=True, metadata={"serializable": True})
+    client: LumaAI = field(
+        default=Factory(
+            lambda self: import_optional_dependency("lumaai").LumaAI(auth_token=self.api_key), takes_self=True
+        ),
+        kw_only=True,
+    )
+    params: dict[str, Any] = field(default={}, kw_only=True, metadata={"serializable": True})
+
+    def try_text_to_video(self, prompt: str) -> VideoArtifact:
+        response = self.client.generations.create(prompt=prompt, **self.params)
+        generation = response
+        status = generation.state
+        while status in ["dreaming", "queued"]:
+            time.sleep(5)
+            if not generation.id:
+                raise Exception("Generation ID not found in the response")
+
+            generation = self.client.generations.get(generation.id)
+            status = generation.state
```

Comment:
try to use the `tenacity` library for this here. i know this is the same approach as the cloud driver but we should try to clean this pattern up

Reply:
I'm actually not sure about this. Retrying (tenacity) feels different than polling (what we're doing here).

Reply:
tenacity has built in conditions for `if_condition` or `if_not_condition`. its not just for retrying on errors

Reply:
I don't doubt it can be used for non-exception things, but all of the examples are for exception retrying which makes me think that's its primary purpose.

Do you have an example of how tenacity might be used for polling?

Reply:
something like
```
from tenacity import retry, retry_if_result, stop_after_attempt, wait_fixed

@retry(
    retry=retry_if_result(lambda result: result is None),
    stop=stop_after_attempt(3),
    wait=wait_fixed(5),
)
def call_api(url_to_poll: str) -> Optional[str]:
    response = requests.get(url_to_poll)
    if response.status_code != 200:
        return None
    return response.text
```